frontend/app/components/card.tsx

```diff
@@ -0,0 +1,182 @@
+import { cn } from '~/utils/tailwind-utils';
+
+type CardBaseProps = Omit<ComponentProps<typeof AppLink>, 'to' | 'file'> & {
+  to?: To;
+  file?: I18nRouteFile;
+};
+
+type CardProps = CardBaseProps & {
+  /**
+   * Optional FontAwesome icon to be displayed.
+   */
+  icon?: IconProp;
+
+  /**
+   * The flip property for the icon.
+   */
+  iconFlip?: FlipProp;
+
+  /**
+   * Optional image URL to be displayed at the top of the card.
+   */
+  image?: string;
+
+  /**
+   * Alternative text for the image.
+   */
+  imageAlt?: string;
+
+  /**
+   * Optional tag text to be displayed at the top-left corner of the card.
+   */
+  tag?: string;
+
+  /**
+   * The title text to be displayed.
+   */
+  title: string;
+
+  /**
+   *  The HTML element to be used for the title
+   *  @default 'h3'
+   */
+  titleAs?: ExtractStrict<HTMLElementType, 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p'>;
+};
+
+/**
+ * Card component renders a card with an optional image, icon, tag.
+ * It uses the AppLink component for navigation and supports various customization options.
+ */
+export function Card({ children, className, icon, iconFlip, image, imageAlt = '', tag, title, titleAs, ...props }: CardProps) {
+  return (
+    <CardBase
+      className={cn(
+        'group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4',
+        'transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90',
+        'aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none',
+        className,
+      )}
+      {...props}
+    >
+      <CardImage image={image} imageAlt={imageAlt} />
+      <div className="relative p-8">
+        <CardTag tag={tag} />
+        <div className="flex items-center gap-4">
+          <CardIcon icon={icon} iconFlip={iconFlip} />
+          <div className="space-y-2">
+            <CardTitle title={title} titleAs={titleAs} />
+            <CardContent>{children}</CardContent>
+          </div>
+        </div>
+      </div>
+    </CardBase>
+  );
+}
+
+/**
+ * Base component for the Card component
+ * Returns an AppLink when the 'To' or 'File' prop is provided.
+ * Otherwise, returns an Anchor tag that functions as a button.
+ */
+function CardBase(props: CardBaseProps) {
+  const { to, file, disabled } = props;
+  const baseClassName = `select-none rounded-sm ${disabled ? 'bg-white' : 'bg-blue-900'}`;
+  if (to || file) {
+    return (
+      <div className={baseClassName}>
+        <AppLink {...(props as ComponentProps<typeof AppLink>)} />
+      </div>
+    );
+  }
+  return (
+    <div className={baseClassName}>
+      <a
+        role="button"
+        aria-disabled={disabled}
+        tabIndex={0}
+        onClick={(e) => {
+          e.preventDefault();
+          props.onClick?.(e);
+        }}
+        onKeyDown={(e) => {
+          e.preventDefault();
+          props.onKeyDown?.(e);
+        }}
```

Comment:
I don't think you have to define `onClick` and `onKeyDown` since you are spreading those `props`.